src/thor/trace_route_action.cc

```diff
@@ -446,6 +446,11 @@ thor_worker_t::map_match(Api& request) {
         // (NOTE that, in between edges may have same edge id with the last edge) right before where
         // discontinuity occurs
         auto prev_match_result = match_result_itr++;
+        if (match_result_itr >= match_results.cend()) {
```

Comment:
we are in the middle of refactoring this, so i  think this is good enough for now until the refactor lands (which will delete all this extremely error-prone code)